doc/architecture/decisions/0014-paginating-large-data-sets.md

```diff
@@ -0,0 +1,161 @@
+pages there are, what the current page is, and the number of the next page etc.
+
+Before we start this work, we should ensure that we agree a common approach to be rolled out
+across all endpoints and get agreement from everyone.
+
+## Decision
+
+We will implement "page-based pagination", starting first with the `/placement-requests/dashboard`
+endpoint. The page query will look as follows:
+
+```
+/placement-requests/dashboard?page=$pageNumber
+```
+
+Where `$pageNumber` is the page number the client wants to return (starting at 1). If the 
+`$pageNumber` is not present, the API returns all results. This can be tweaked as the 
+service evolve, and we have pagination on all endpoints.
+
+This will then be passed into the service to determine which page of results to return.
+
+We could possibly use a [`PagingAndSortingRepository`][1] for this.
+
+As well as returning the data in the body, we will also need to return metadata about the 
+pagination. To ensure compatibility and also make frontend integration easier, we should
+return this information in the header with the following information:
+
+- The current page
+- The total number of pages
+- The total number of results
+- The page size (this will be a default value, but useful for compatibility)
+
+This could be implemented in the OpenAPI spec like so:
+
+```yml
+  /placement-requests/dashboard:
+    get:
+      tags:
+        - Placement requests
+      summary: Gets all placement requests
+      parameters:
+        - name: isParole
+          in: query
+          description: States whether or not to return parole cases
+          schema:
+            type: boolean
+      responses:
+        200:
+          description: successfully retrieved placement requests
+          headers:
+            X-Pagination-CurrentPage:
+              schema:
+                type: integer
+              description: The current page number
+            X-Pagination-TotalPages:
+              schema:
+                type: integer
+              description: The total number of pages
+            X-Pagination-TotalResults:
+              schema:
+                type: integer
+              description: The total number of results
+            X-Pagination-PageSize:
+              schema:
+                type: integer
+              description: The total number of results
+          content:
+            'application/json':
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/PlacementRequest'
+        401:
+          $ref: '#/components/responses/401Response'
+        403:
+          $ref: '#/components/responses/403Response'
+        500:
+          $ref: '#/components/responses/500Response'
+```
+
+Again, using the `PagingAndSortingRepository`, we can get this metadata from the response and return
+it alongside the actual data itself, perhaps using a `Pair`, e.g:
```

Comment:
is it worth including a link to some docs on `Pair` for the benefit of Kotlin newbies? 

e.g. https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/